backend/corpora/common/entities/collection.py

```diff
@@ -344,6 +351,34 @@ def update(self, links: list = None, keep_links=False, **kwargs) -> None:
 
         super().update(**kwargs)
 
+    def update_curation(self, links: list = None, keep_links=False, **kwargs) -> None:
+        """
+        Update an existing collection to match provided the parameters. The specified columns are replaced. The DOI is
+        handled as a link for legacy reasons (for consistency with Corpora API), but independently of the other links.
+        :param links: links to create and connect to the collection. If present, the existing attached entries will
+         be removed and replaced with new entries.
+        :param keep_links: boolean - whether or not links need to be preserved. Links are preserved if True.
+        :param kwargs: Any other fields in the dataset that will be replaced.
+        """
+        new_links = links if links else []
+
+        if not keep_links:
+            for old_link in self.links:
+                if old_link.link_type != ProjectLinkType.DOI:
+                    self.session.delete(old_link)
+
+        for new_link in new_links:
+            if new_link["link_type"] == ProjectLinkType.DOI.name:
+                # A new DOI is being introduced so we delete existing DOI(s)
+                for old_link in self.links:
+                    if old_link.link_type == ProjectLinkType.DOI:
+                        self.session.delete(old_link)
```

Comment:
I change this code a bit so flagging in case you want to reinstate your previous comment @Bento007